### ISSUE TITLE
JSON AST encoding for a cycle over mapping type

### ISSUE THREAD

**Issue by unboxedtype** [everx-labs/TVM-Solidity-Compiler#58]
Hi!

The sample is as follows:
```
pragma ton-solidity ^0.44.0;

contract Sample {
    struct Info {
        uint id;
        uint val;
    }
    mapping (address => Info) public _addrs;
    
    function calcsum() external view returns (uint) {
        uint sum;
        for ((, Info oi) : _addrs) {
            sum += oi.val;
        }
        return sum;
    }
}
```

I compile it using 
```bash
$ solc --compact-ast-json ./Sample.sol
```
The resulting JSON file map the ``for`` statement into an ordinary ``ExpressionStatement``, without any clue regarding
the nature of this statement, i.e. cycle over a mapping. 

Is it intentionally done this way?

I would expect it to be at least a ``ForStatement``. It would be even better to distinguish it with a special ``ForOverMappingStatement``, because it has different syntax/AST nodes.

**Comment by IgorKoval** [comment]
Thank you for your report. We are solving this problem.

**Comment by BorisI** [comment]
A number of extended AST JSON info generation issues addressed in 0.47. Feedback on the completeness, correctness and usability of the data supplied is most welcome,

**Comment by IgorKoval** [comment]
It has been fixed.
